pkg/apis/azure/validation/shoot.go

```diff
@@ -158,3 +164,28 @@ func validateVolumeFunc(volumeType *string, volumeSize string, encrypted *bool,
 	}
 	return allErrs
 }
+
+// ValidateUpgradeV120ToV121 prevents that Shoots get updated from k8s v1.20 to v1.21.
+// TODO(dkistner) remove this once csi-driver async operation issues are resolved.
+func ValidateUpgradeV120ToV121(newShoot, oldShoot *core.Shoot, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	oldShootVersion, err := semver.NewVersion(oldShoot.Spec.Kubernetes.Version)
+	if err != nil {
+		return append(allErrs, field.Invalid(fldPath, oldShoot.Spec.Kubernetes.Version, "could not parse kubernetes version"))
+	}
+
+	newShootVersion, err := semver.NewVersion(newShoot.Spec.Kubernetes.Version)
+	if err != nil {
+		return append(allErrs, field.Invalid(fldPath, newShoot.Spec.Kubernetes.Version, "could not parse kubernetes version"))
+	}
+
+	if oldShootVersion.LessThan(k8sV121) && (newShootVersion.Equal(k8sV121) || newShootVersion.GreaterThan(k8sV121)) {
+		if value, ok := newShoot.Annotations[azure.ForceK8sVersionUpgrade]; ok && value == "true" {
+			return allErrs
+		}
+		allErrs = append(allErrs, field.Forbidden(fldPath, "upgrade Kubernetes version from < v1.21 to >= v1.21.0 are disabled"))
```

Comment:
I think the error message should mention the workaround annotation. It is. more convenient for the user or maybe even the provider-extension issue for more information.